core/src/main/java/org/elasticsearch/index/query/MatchNoneQueryParser.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+public class MatchNoneQueryParser extends BaseQueryParser {
```

Comment:
I thought this query would be useful to users as well but looks like we prefer to keep it internal only. In that case we can remove this parser class, it will be similar to EmptyQueryBuilder. A query that can be streamed but has no corresponding parser. We register it via `namedWriteableRegistry.registerPrototype(QueryBuilder.class, MatchNoneQueryBuilder.PROTOTYPE);` in `IndicesQueriesRegistry`, like we do with `EmptyQueryBuilder`.